autowarefoundation/autoware.universe
# fix(system_monitor): fix parameter threshold of CPU Usage monitoring

Signed-off-by: v-nakayama7440-esol <v-nakayama7440@esol.co.jp>

## Description

Change CPU usage threshold and counter threshold values because there is delay to generate a warning/error after the fix https://github.com/autowarefoundation/autoware.universe/pull/557. There is regression.
- Before the fix:
  - ≧96% x 1 time -> Warning, ≧96% x consecutive 2 times -> Error
  - ≧100% x 1 time -> Error
- After the fix:
  -  ≧96% x consecutive 2 times -> Warning
  - ≧100% x consecutive 2 times -> Error

I believe 100% of CPU usage at a moment shall be allowed. Or rather continuous CPU high load will possibly cause hazardous conditions.
- So the change is made as follows:
  - ≧96% x 1 time -> Warning, ≧96% x consecutive 2 times -> Error

## Related links

https://github.com/autowarefoundation/autoware.universe/pull/557

## Tests performed

1. Simulate CPU high load by using stress-ng command.
   ```console
   stress-ng -k -c 12 -l 96
   ```

1. Verify `cpu_monitor` generates a warning at 1 time of ≧96%. 
1. Verify `cpu_monitor` generates an error  at consecutive 2 times of ≧96%. 
   ```console
   ros2 topic echo /diagnostics | grep -e "cpu_monitor: CPU Usage" -A 7
   ```
   <img width="527" alt="test" src="https://user-images.githubusercontent.com/97144416/189017218-dccec268-3b86-4312-8041-17841cb1151c.png">

## Notes for reviewers



## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
